Header header
string input
string type